### DISCUSSION TITLE
Release source code if development is halted?

### DISCUSSION THREAD

**Discussion by digitalirony** [AnthonyMusgrove/Emby-ScripterX#60] [Q&A] [answered]
You have events for onSessionStarted and OnSessionEnded....but you don't expose %session.id% so using /Session/id/ api calls isn't the easiest. 

Also, your website and wiki are both broken.

**Comment by digitalirony** [comment]
@AnthonyMusgrove Sorry meant to tag you in the main post. But adding it here now.

**Comment by AnthonyMusgrove** [comment, marked as answer]
Hi @digitalirony,

Thank you for your suggestion - please note that I have added %session.id% to all session events, and will be available in the next release.  Please note however, the next release is aimed at the Emby-Server Beta 4.7+ and will not function correctly on server versions below this.

Thank you again mate, I appreciate it very much!

                   (4.0.0.1 release): 
                    added : { "token": "%session.id%", "desc": "The id of the session being used to trigger the session-event." }

**Comment by AnthonyMusgrove** [comment]
![image](https://user-images.githubusercontent.com/55648663/163717830-13b45e55-900a-49cd-9914-5c50bbba8a3a.png)

**Comment by AnthonyMusgrove** [comment]
Session STARTED with ID c8be66f7bb87acee3136d4da6d0f7a45 for user anthony on device Microsoft 
Session ENDED with ID c8be66f7bb87acee3136d4da6d0f7a45 for user anthony on device Microsoft 

session.bat script:
REM Parameters expected:  <action:started/ended> %session.id% %username% %device.name% 
@echo Session %1 with ID %2 for user %3 on device %4 >> C:\embyscripts\session.log

**Comment by digitalirony** [comment]
@AnthonyMusgrove What about wiki/community scripts? 
I am just trying to use it to send people Alert messages when they start a new session about upcoming maintenance. 
This seems like the best way to donthat across all client types.
Can you use these env variables in the post and get methods in your module?

**Reply by AnthonyMusgrove** [reply]
Yep, can help you with this shortly mate.  Just rectifying some core code for LiveTV, won't be too long!

**Reply by AnthonyMusgrove** [reply]
I added something SPECIFICALLY for your usecase for you -- a great idea that you've helped bring to ScripterX, thank you!

![image](https://user-images.githubusercontent.com/55648663/163922831-e1a79cc8-7ad8-457a-8d91-dae212de5832.png)

**Reply by digitalirony** [reply]
@AnthonyMusgrove This is awesome. I am currently on Version  4.7.0.33 beta, so what can I do to use these new features?

**Reply by AnthonyMusgrove** [reply]
Thank you so much @digitalirony !  The latest beta version is 4.0.0.2 which I have released (just now).  You can download and install it via the official Emby Plugin catalog, just make sure you choose the Beta version 4.0.0.2 at the top of the list.  **NOTE THOUGH - It will only work for Beta version 4.7.0.35 and above, I notice you're on .33, so you'll probably need to update first! **

![image](https://user-images.githubusercontent.com/55648663/163939870-f1de0b4e-35cc-420a-b380-5069e2d66be2.png)

**Comment by AnthonyMusgrove** [comment]
4.0.0.1 BETA is on the catalog now for testing, comments & suggestions.  Still working mainly on LiveTV components.

![image](https://user-images.githubusercontent.com/55648663/163809128-9b098c1f-3c5b-40db-a879-e1be9106b130.png)

**Comment by digitalirony** [comment]
@AnthonyMusgrove I am trying this out now.

I am only asking here, because I am unaware of any other channel to ask you. But, is there any chance of you making a plugin for emby to allow for a custom 'channel' or 'reel'?

For instance, on 4/20(yesterday), it would have been cool for me to curate a Stoner Movie collection to spotlight for my users.
Or maybe I just want to create a spotlight channel for other reasons etc. It seems something a lot of people would like.

**Comment by digitalirony** [comment]
Ohhh. Cool. So far this session start message is great! Thanks so much!
